tests/check_check_sub.c

```diff
@@ -497,6 +497,284 @@ START_TEST(test_ck_assert_uint_expr)
+  int d = 2;
+  int f = 2;
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_lt(3%d, 2%f);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_le)
+{
+  record_test_name(tcase_name());
+
+  double x = 2.0;
+  double y = 2.5;
+  ck_assert_floating_le(x, y);
+  ck_assert_floating_le(x, x);
+  y-=1;
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_le(x, y);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_le_with_mod)
+{
+  record_test_name(tcase_name());
+
+  int d = 2;
+  int f = 2;
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_le(3%d, 2%f);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_gt)
+{
+  record_test_name(tcase_name());
+
+  double x = 2.5;
+  double y = 2.0;
+  ck_assert_floating_gt(x, y);
+  y+=1;
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_gt(x, y);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_gt_with_mod)
+{
+  record_test_name(tcase_name());
+
+  int d = 2;
+  int f = 2;
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_gt(2%d, 3%f);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_ge)
+{
+  record_test_name(tcase_name());
+
+  double x = 2.5;
+  double y = 2.0;
+  ck_assert_floating_ge(x, y);
+  ck_assert_floating_ge(x, x);
+  y+=1;
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_ge(x, y);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_ge_with_mod)
+{
+  record_test_name(tcase_name());
+
+  int d = 2;
+  int f = 2;
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_ge(2%d, 3%f);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_eq_prec)
+{
+  record_test_name(tcase_name());
+
+  double x = 0.0001;
+  double y = 0.0002;
+  ck_assert_floating_eq_prec(x, y, 3);
+  ck_assert_floating_eq_prec(x, x, 3);
+  x*=10;
+  y*=10;
+  ck_assert_floating_eq_prec(x, y, 2);
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_eq_prec(x, y, 3);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_ne_prec)
+{
+  record_test_name(tcase_name());
+
+  double x = 0.0001;
+  double y = 0.0002;
+  ck_assert_floating_ne_prec(x, y, 4);
+  x*=10;
+  y*=10;
+  ck_assert_floating_ne_prec(x, y, 3);
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_ne_prec(x, y, 2);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_eq_prec_with_mod)
+{
+  record_test_name(tcase_name());
+
+  int d = 2;
+  int f = 2;
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_eq_prec(3%d, 2%f, 1);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_eq_prec_min_prec)
+{
+  record_test_name(tcase_name());
+
+  int min_prec = 0;
+  ck_assert_floating_eq_prec(0, 0, min_prec);
+  int min_prec_minus_one = min_prec - 1;
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_eq_prec(0, 0, min_prec_minus_one);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_eq_prec_max_prec)
+{
+  record_test_name(tcase_name());
+
+  int max_prec = -LDBL_MIN_10_EXP;
+  ck_assert_floating_eq_prec(0, 0, max_prec);
+  // TODO: test with -(LDBL_MIN_10_EXP - 1) and platform specific output.
```

Comment:
Well, meaning of precision is number of digits after decimal point. So We pass this number to all functions. But it is negative power of ten. So we need to test exactly against LDBL_MIN_10_EXP. Testing against LDBL_MAX_10_EXP would mean that I could use ten in positive power. But it has no sense in context of digits after decimal point.